circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit_inner.cpp

```diff
@@ -137,6 +137,7 @@ KernelCircuitPublicInputs<NT> native_private_kernel_circuit_inner(DummyComposer&
 
     // TODO(jeanmon) Resuscitate after issue 499 is fixed as explained below.
     // validate_this_private_call_hash(composer, private_inputs, public_inputs);
+    array_pop(public_inputs.end.private_call_stack);
```

Comment:
@jeanmon mind taking a look at this part?

Reply:
@spalladino This change is ok as the commented function should have done this step. Could you please add a small comment to make it clear that this line must be removed once we uncomment the validate_this_private_call_hash?